scipy/special/tests/test_basic.py

```diff
@@ -1326,17 +1326,13 @@ def test_comb(self):
     @pytest.mark.parametrize("N", [4.5, 4])
     def test_comb_legacy(self, N, k, legacy, repetition):
         # test is only relevant for exact=True
-        if legacy and (N != int(N) or k != int(k)):
+        if legacy:
             with pytest.warns(
                 DeprecationWarning,
-                match=r"Non-integer arguments are currently being cast to",
+                match="Using 'legacy=True' is deprecated",
             ):
                 result = special.comb(N, k, exact=True, legacy=legacy,
                                       repetition=repetition)
-        else:
-            result = special.comb(N, k, exact=True, legacy=legacy,
-                                  repetition=repetition)
-        if legacy:
```

Comment:
Aside from the change in the warning, this test shouldn't have to change AFAICT.